### DISCUSSION TITLE
Calling `end()` doesn't release IPs.

### DISCUSSION THREAD

**Discussion by the-maldridge** [earlephilhower/arduino-pico#2630] [Q&A]
I have both a wireless and wired interface setup, and they operate in an exclusive configuration where if the wired interface has link it forces the wireless interface down.  When this happens, the wireless interface seems to retain its IP address, which is unexpected for a downed netdev.  Is this intentional?  If so is there any way to get rid of an IP address once its been assigned?  I tried calling `config(IPAddress())` to try to get back to an unset address, but this doesn't seem to have an effect.

**Comment by earlephilhower** [comment]
LWIP doesn't send a DHCP release when a netif is removed, if that's what you mean.  So your DHCP server log will still show an active lease on the downed interface.

If it's not that, can you make a small MCVE that shows this?  Without a netif, LWIP isn't going to know about any IPs but there may be state further up the stack that's inconsistent in that case.

**Reply by the-maldridge** [reply]
Sure, what I am referring to isn't DHCP.  I'm fairly confident nobody's implementation reliably sends a DHCP release event.  What I'm referring to is that after calling `WiFi.end()` a subsequent call to `WiFi.localIP()` still returns the last address configured.  This doesn't seem quite right since the IP should not have been retained by a stopped netif.

I'm not at a full sized computer at the moment, but I should be able to draft a sample program this weekend that demonstrates the behavior.

**Reply by earlephilhower** [reply]
Gotcha, and it makes sense now.  See:  https://github.com/earlephilhower/arduino-pico/pull/1866

This is actually intended be this way as part of the Arduino network "standardization" efforts that @JAndrassy was working hard on last year (on this and other projects).

**Reply by the-maldridge** [reply]
I see.  I'll give `config(INADDR_NONE)` a try and see how that works out.  The problem I'm working around is that the DHCP server is configured to hand out the same address to both wired and wireless interfaces, but this results in a duped address being present in the LWIP routing table that, of course, does not work.

**Reply by the-maldridge** [reply]
In testing, neither `config(INADDR_NONE)` nor than `config(0)` causes the IP to be removed.  Is there something else that needs to happen to cycle the IP?  I tried doing an end/begin cycle after changing config, but that seemed to have no effect.

**Reply by the-maldridge** [reply]
On further testing, I think this might be a misunderstanding in what I'm expecting to happen.  I was expecting that on setting `config(0)` that the IP returned by a call to `localIP()` would yield an unset IP, but this does not appear to be happening.

As an aside, the W5500 driver seems to require a hard delay of at least 5ms when not operating with interrupts.  I could not find this documented anywhere and determined this value experimentally, perhaps it should get added to the ethernet page in the docs?
